src/AiidaDFTK.jl

```diff
@@ -101,7 +101,24 @@ Run a DFTK calculation from a json input file.
 Output is by default written to `stdout` and `stderr`.
 The list of generated output files is returned.
 """
-function run_json(filename::AbstractString; extra_output_files=String[])
+function run_json(filename::AbstractString; extra_output_files=String[], min_version=typemin(VersionNumber), max_version=typemax(VersionNumber))
+    @info("$LOG_IMPORTS_SUCEEDED --"
+        * " This indicates that AiidaDFTK was installed correctly"
+        * " and that the MPI environment is likely correct.")
+
+    version = pkgversion(@__MODULE__)
+    if min_version <= version && version < max_version
+        @info("$LOG_VERSION_OK --"
+            * " Expected AiidaDFTK version ∈ [$min_version, $max_version)."
+            * " Actual: $(version).")
+    else
+        msg = ("$LOG_VERSION_MISMATCH --"
+            * " Expected AiidaDFTK version ∈ [$min_version, $max_version)."
+            * " Actual: $(version).")
+        @error msg
+        error(msg)
```

Comment:
This is weird ... I see why you do it, but it just feels wrong. Is there a cleaner way without duplicating the message on the terminal ?

Reply:
I managed to catch exceptions and log them, it's a lot nicer that way I think. :)